sdl_viewer/shaders/terrain.vs

```diff
@@ -0,0 +1,55 @@
+#version 410 core
+// The vertex positions are in grid coordinates, i.e. between 0 and grid_size (inclusive).
+layout(location = 0) in ivec3 aPos;
+
+// These are constant throughout the program.
+uniform double grid_size;
+uniform dvec3 terrain_origin_m;
+uniform double terrain_res_m;
+
+// The grid coordinates of the min corner of the currently visible terrain (the window).
+// Essentially, the terrain regular grid with resolution terrain_res_m and origin
+// terrain_origin_m, and the mesh is rendering a rectangle within that grid.
+uniform dvec2 terrain_pos;
+// These two are equal and set by the GlMovingWindowTexture.
+// They hold the coordinates (in pixels) of the min corner
+// of the texture within the sampler.
+uniform ivec2 height_texture_offset;
+uniform ivec2 color_texture_offset;
+// The full projective transform.
+uniform dmat4 world_to_gl;
+// The frame of the terrain.
+// TODO(nnmm): This can also include the terrain_origin.
+uniform dmat4 terrain_to_world;
+// The textures.
+uniform sampler2D height;
+uniform sampler2D color;
+
+// The data that's passed to the geometry shader.
+out VS_OUT {
+  vec4 color;
+  uint quads;
+} vs_out;
+
+void main() {
+  // Each vertex position (aPos) maps to a pixel in the texture, but
+  // shifted (by height_texture_offset) with wraparound.
+  ivec2 texCoord = height_texture_offset + aPos.xy;
+  ivec2 texSize = textureSize(height, 0);
+  ivec2 texCoordModSize = texCoord - (texCoord / texSize) * texSize;
```

Comment:
can we use https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/mod.xhtml here?

Reply:
That's for floats, unfortunately

Reply:
i see